### PR TITLE
Migrate "User.GetProfilesWithoutTeam" to Sync by default

### PR DESCRIPTION
**Pull request by himynameisdave**
#### Summary
This PR migrates `User.GetProfilesWithoutTeam` to Sync by default.

#### Ticket Link
https://github.com/mattermost/mattermost-server/issues/11458

#### Notes

I am still very new to Golang, please let me know if anything doesn't conform to proper style, as I am very eager to learn (and to contribute to Mattermost)!

### CODE REVIEW

File: app/user.go [hunk 1]
```diff
@@ -583,11 +583,11 @@ func (a *App) GetUsersWithoutTeamPage(page int, perPage int, asAdmin bool, viewR
 }
 
 func (a *App) GetUsersWithoutTeam(offset int, limit int, viewRestrictions *model.ViewUsersRestrictions) ([]*model.User, *model.AppError) {
-	result := <-a.Srv.Store.User().GetProfilesWithoutTeam(offset, limit, viewRestrictions)
```
Comment (jespino) [inline app/user.go]: You can simplify this with just `return a.Srv.Store.User().GetProfilesWithoutTeam(offset, limit, viewRestrictions)`

File: store/storetest/user_store.go [hunk 2]
```diff
@@ -886,21 +886,21 @@ func testUserStoreGetProfilesWithoutTeam(t *testing.T, ss store.Store) {
...
+		result, err := ss.User().GetProfilesWithoutTeam(1, 1, nil)
+		require.Nil(t, err)
+		assert.Equal(t, []*model.User{sanitized(u3)}, result)
 	})
 
 	t.Run("get, offset 2, limit 1", func(t *testing.T) {
-		result := <-ss.User().GetProfilesWithoutTeam(2, 1, nil)
-		require.Nil(t, result.Err)
-		assert.Equal(t, []*model.User{}, result.Data.([]*model.User))
+		result, nil := ss.User().GetProfilesWithoutTeam(2, 1, nil)
```
Comment (jespino) [inline store/storetest/user_store.go]: you are putting here `nil` instead of  `err`

File: store/storetest/user_store.go [hunk 3]
```diff
@@ -886,21 +886,21 @@ func testUserStoreGetProfilesWithoutTeam(t *testing.T, ss store.Store) {
 	defer func() { require.Nil(t, ss.Bot().PermanentDelete(u3.Id)) }()
 
 	t.Run("get, offset 0, limit 100", func(t *testing.T) {
-		result := <-ss.User().GetProfilesWithoutTeam(0, 100, nil)
-		require.Nil(t, result.Err)
-		assert.Equal(t, []*model.User{sanitized(u2), sanitized(u3)}, result.Data.([]*model.User))
+		result, err := ss.User().GetProfilesWithoutTeam(0, 100, nil)
```
Comment (jespino) [inline store/storetest/user_store.go]: Probably we can rename `result` to something more meaningful, like `users`

**Comment by jespino** [comment]
@himynameisdave Another thing, you need to execute `make store-mocks` to update the mocks with your changes.

File: store/storetest/user_store.go (hunk 2 above)
Comment (himynameisdave) [reply store/storetest/user_store.go]: Whoops!

**Comment by himynameisdave** [comment]
@jespino I've addressed your feedback, please let me know if you need me to make anymore changes 👍

**Pull request merged**